### PR TITLE
Black market ammo shop category + new ammo boxes

### PR DESCRIPTION
**Pull request by Blundir**
# About the pull request

Adds dedicated ammo category to black market, it has all the ammo types of guns that are available on black market already.

I mostly took prices of corresponding weapon kits, doubled amount of ammo, doubled the price, which is fair imo, since you do not get the gun, but pay that price for practicality of packaging.
In cases of ammo boxes, just used the same logic and multipled price by amount of ammo in ammo box.

Deagle High Impact Armor-Piercing ammo box is empty, honk.

Added a lot of new ammo boxes for all the ammo that is in the black market, also added some additional ones (mar30 ex, mar50, they are unavailable in req, I just added boxes), also added craft recipe to all of them.

# Explain why it's good for the game

Balance wise, i didn't add anything new, i didn't add nsg ap/ex or type 19 ex or so on, i didn't make it cheaper to shoot those guns, you can already just order weapon kits and gather ammo from there, and it costs even a bit cheaper, but it's just less convenient and more frustrating to buy 10 same kits when you just want the ammo, also buying a lot of kits creates a lot of contraband guns in general, which isn't good.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

aFLhIk7Vfv

dreamseeker_c6Xo5oHFvn

dreamseeker_hccBNDu8rA

</details>

# Changelog
:cl:
add: contraband ammo selection in black market
add: tons of new ammo boxes
/:cl:

### CODE REVIEW

**Comment by vero5123** [comment]
bazooka when

**Comment by Blundir** [comment]
> bazooka when

i'm not suicidal, yet

**Review by harryob — APPROVED** [review]

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -328,6 +328,34 @@ GLOBAL_LIST_INIT_TYPED(cardboard_recipes, /datum/stack_recipe, list ( \
...
+		null, \
+		new/datum/stack_recipe("empty magazine box (FN FP9000)", /obj/item/ammo_box/magazine/fp9000/empty), \
+		null, \
+		new/datum/stack_recipe("empty magazine box (Type19)", /obj/item/ammo_box/magazine/type19/empty), \
+		null, \
+		new/datum/stack_recipe("empty magazine box (ZhNK-72)", /obj/item/ammo_box/magazine/zhnk/empty), \
+		null, \
+		new/datum/stack_recipe("empty magazine box (Type64 Bizon)", /obj/item/ammo_box/magazine/type64/empty), \
+		null, \
+		new/datum/stack_recipe("empty magazine box (S&W .38)", /obj/item/ammo_box/magazine/snw/empty), \
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:357]: someday this really needs to be redone to be not-terrible

File: code/modules/projectiles/ammo_boxes/magazine_boxes.dm [hunk 2]
```diff
@@ -735,3 +735,297 @@
 
 /obj/item/ammo_box/magazine/deagle/super/highimpact/ap/empty
 	empty = TRUE
+
+
+//-----------------------S&W Revolver Speed Loaders Box-----------------------
+/obj/item/ammo_box/magazine/snw
+	name = "\improper speed loaders box (S&W .38 x 12)"
```
Comment (Zonespace27) [inline code/modules/projectiles/ammo_boxes/magazine_boxes.dm:742]: \improper is for when the first letter is capitalized but it's not a proper noun

**Review by Zonespace27 — DISMISSED** [review]

**Pull request merged**